src/HttpOverStream.Server.Owin/Once.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace HttpOverStream.Server.Owin
+{
+    public class Once<T>
+    {
+        private Func<T> _todo;
+        public Once(Func<T> todo)
+        {
+            _todo = todo;
+        }
+
+        public (T, bool) Do()

```

Comment:
I think the 'maybe do a thing' pattern in c# uses naming with `Try` so this could be `TryDo`.

I think the bool should probably come first in the output, following things like TryGetValue on dictionary, you get the bool first and foremost and then you only care about the `out` param if the bool is true.

Reply:
IIRC, the `TryGet` approach has been deprecated since the introduction of tuples. (I might be wrong though)

Reply:
I think the `out` param is deprecated, i'm not sure about the naming, i'd be interested to see examples. 

Either way, calling it `Do` when it may or may not actually do it, might be improved with a more 'maybe' sounding name.

Not a big deal either way, though